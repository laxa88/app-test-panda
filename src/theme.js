import React from 'react';

export const panda = {
  primaryColor: '#33A00E',
};

export const sakura = {
  primaryColor: 'pink',
};

export const ThemeContext = React.createContext({
  theme: panda,
});
