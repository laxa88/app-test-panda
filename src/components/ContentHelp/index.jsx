import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import styled from 'styled-components';
import Accordion from '../Accordion';

import { currentTheme } from '../../theme';

export const StyledBody = styled.div`
  padding: 20px;

  & svg {
    color: ${props => props.theme.secondaryColor2};
    font-size: 110px;
    float: left;
    padding: 0 20px 20px 0;
  }

  & p {
    font-size: 14px;
  }
`;

const Body = (
  <StyledBody theme={currentTheme}>
    <FontAwesomeIcon icon={['far', 'life-ring']} />

    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec
      faucibus lacus, ut aliquet est. Sed et elit ex. Vestibulum nisl nisi,
      aliquam id justo sit amet, varius consequat diam. Vivamus at dui ornare
      enim ultricies sollicitudin ut non orci. Integer sagittis sapien et nisl
      maximus varius. In vitae maximus nulla, a tempus eros.
    </p>

    <p>
      Pellentesque habitant morbi tristique senectus et netus et malesuada fames
      ac turpis egestas. Phasellus ac magna vehicula enim vestibulum finibus sed
      ut sem. In hac habitasse platea dictumst. Donec porttitor nisi vitae quam
      dapibus, vel posuere purus mollis. Integer ut rhoncus odio.
    </p>

    <p>
      Donec porttitor nisi vitae quam dapibus, vel posuere purus mollis. Integer
      ut rhoncus odio. Nulla sit amet facilisis dolor, sit amet sagittis ante.
      Nam non felis in ipsum lacinia blandit.
    </p>
  </StyledBody>
);

const ContentHelp = props => <Accordion header="Help" body={Body} {...props} />;

export default ContentHelp;
