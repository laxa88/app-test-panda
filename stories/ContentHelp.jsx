/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';

import ContentHelp from '../src/components/ContentHelp';
import { sakura, ThemeContext } from '../src/theme';

storiesOf('ContentHelp', module)
  .add('default', () => <ContentHelp />)
  .add('sakura', () => (
    <ThemeContext.Provider value={{ theme: sakura }}>
      <ContentHelp />
    </ThemeContext.Provider>
  ));
