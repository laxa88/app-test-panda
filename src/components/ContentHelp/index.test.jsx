import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ContentHelp, { Body, StyledBody } from './index';
import { currentTheme } from '../../theme';

describe('<ContentHelp />', () => {
  it('renders', () => {
    const result = renderer.create(<ContentHelp />);

    expect(result.root.children[0].props.children.props.header).toBe('Help');
  });

  it('renders Body', () => {
    const result = renderer.create(Body);

    expect(result.toJSON()).toMatchSnapshot();
  });

  it('renders StyledBody', () => {
    const result = renderer.create(<StyledBody theme={currentTheme} />);

    expect(result.toJSON()).toHaveStyleRule(
      'color',
      currentTheme.secondaryColor2,
      { modifier: '& svg' },
    );
  });
});
