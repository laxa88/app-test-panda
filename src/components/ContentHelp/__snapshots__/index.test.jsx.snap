// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentHelp /> renders Body 1`] = `
.c0 {
  padding: 20px;
}

.c0 svg {
  color: #999999;
  font-size: 110px;
  float: left;
}

.c0 p {
  font-size: 14px;
}

<div
  className="c0"
>
  Mocked_FontAwesomeIcon (far,life-ring)
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec faucibus lacus, ut aliquet est. Sed et elit ex. Vestibulum nisl nisi, aliquam id justo sit amet, varius consequat diam. Vivamus at dui ornare enim ultricies sollicitudin ut non orci. Integer sagittis sapien et nisl maximus varius. In vitae maximus nulla, a tempus eros.
  </p>
  <p>
    Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Phasellus ac magna vehicula enim vestibulum finibus sed ut sem. In hac habitasse platea dictumst. Donec porttitor nisi vitae quam dapibus, vel posuere purus mollis. Integer ut rhoncus odio.
  </p>
  <p>
    Donec porttitor nisi vitae quam dapibus, vel posuere purus mollis. Integer ut rhoncus odio. Nulla sit amet facilisis dolor, sit amet sagittis ante. Nam non felis in ipsum lacinia blandit. Nulla sit amet facilisis dolor, sit amet sagittis ante. Nam non felis in ipsum lacinia blandit. Curabitur rhoncus, elit sed faucibus malesuada, arcu nisi sollicitudin eros, in porttitor lorem risus nec nunc.
  </p>
</div>
`;
